petab/problem.py

```diff
@@ -146,18 +153,42 @@ def from_files(sbml_file: str = None,
                        visualization_df=visualization_df)
 
     @staticmethod
-    def from_yaml(yaml_config: Union[Dict, str]) -> 'Problem':
+    def from_yaml(yaml_config: Union[Dict, Path, str]) -> 'Problem':
         """
         Factory method to load model and tables as specified by YAML file.
 
         Arguments:
             yaml_config: PEtab configuration as dictionary or YAML file name
         """
-        if isinstance(yaml_config, str):
-            path_prefix = os.path.dirname(yaml_config)
+        if isinstance(yaml_config, (str, Path)):
+            if isinstance(yaml_config, Path):
+                path_prefix = os.path.dirname(yaml_config)
+                get_path = lambda filename:\
+                    f"{path_prefix}{os.sep}{filename}"  # noqa: E731
+            else:
+                # yaml_config may be path or URL
+                path_url = urlparse(yaml_config)
+                if not path_url.scheme or \
+                        (path_url.scheme != 'file' and not path_url.netloc):
+                    # a regular file path string
+                    path_prefix = os.path.dirname(yaml_config)
+                    get_path = lambda filename: \
+                        f"{path_prefix}{os.sep}{filename}"  # noqa: E731
+                else:
+                    # a URL
+                    # extract parent path from
+                    url_path = unquote(urlparse(yaml_config).path)
+                    parent_path = str(PurePosixPath(url_path).parent)
+                    path_prefix = urlunparse(
+                        (path_url.scheme, path_url.netloc, parent_path,
+                         path_url.params, path_url.query, path_url.fragment)
+                    )
+                    # need "/" on windows, not "\"
+                    get_path = lambda filename: \
+                        f"{path_prefix}/{filename}"  # noqa: E731
```

Comment:
All three of these code blocks for `Path`, `str` and `URL` seem quite similar, perhaps they can be combined converting them a type that supports the `/` operator then just define `get_path` once?

Reply:
No, this is unfortunately not possible, because we need different path separators. Thanks to Windows. The previous version was more concise, but didn't work on Windows...